src/qt/nametablemodel.cpp

```diff
@@ -0,0 +1,566 @@
+#include "nametablemodel.h"
+
+#include "guiutil.h"
+#include "walletmodel.h"
+#include "guiconstants.h"
+#include "wallet/wallet.h"
+#include "ui_interface.h"
+#include "platformstyle.h"
+
+#include "names/common.h"
+#include "util.h"
+#include "protocol.h"
+#include "rpc/server.h"
+#include "validation.h" // cs_main
+
+#include <univalue.h>
+
+#include <QDebug>
+#include <QObject>
+#include <QTimer>
+
+// ExpiresIn column is right-aligned as it contains numbers
+namespace {
+    int column_alignments[] = {
+        Qt::AlignLeft|Qt::AlignVCenter,     // Name
+        Qt::AlignLeft|Qt::AlignVCenter,     // Value
+        Qt::AlignRight|Qt::AlignVCenter     // Expires in
+    };
+}
+
+struct NameTableEntryLessThan
+{
+    bool operator()(const NameTableEntry &a, const NameTableEntry &b) const
+    {
+        return a.name < b.name;
+    }
+    bool operator()(const NameTableEntry &a, const QString &b) const
+    {
+        return a.name < b;
+    }
+    bool operator()(const QString &a, const NameTableEntry &b) const
+    {
+        return a < b.name;
+    }
+};
+
+// Returns true if new height is better
+bool NameTableEntry::CompareHeight(int nOldHeight, int nNewHeight)
+{
+    if(nOldHeight == NAME_NON_EXISTING)
+        return true;
+
+    // We use optimistic way, assuming that unconfirmed transaction will eventually become confirmed,
+    // so we update the name in the table immediately. Ideally we need a separate way of displaying
+    // unconfirmed names (e.g. grayed out)
+    if(nNewHeight == NAME_UNCONFIRMED)
+        return true;
+
+    // Here we rely on the fact that dummy height values are always negative
+    return nNewHeight > nOldHeight;
+}
+
+// Private implementation
+class NameTablePriv
+{
+public:
+    CWallet *wallet;
+    QList<NameTableEntry> cachedNameTable;
+    NameTableModel *parent;
+
+    NameTablePriv(CWallet *wallet, NameTableModel *parent):
+        wallet(wallet), parent(parent) {}
+
+    void refreshNameTable()
+    {
+        qDebug() << "NameTableModel::refreshNameTable";
+        cachedNameTable.clear();
+        std::map< std::string, NameTableEntry > vNamesO;
+
+        // unconfirmed (name_pending) names
+        JSONRPCRequest namePendingRequest;
+        namePendingRequest.strMethod = "name_pending";
+        namePendingRequest.params = NullUniValue;
+        namePendingRequest.fHelp = false;
+        UniValue pendingNames;
+
+        try {
+            pendingNames = tableRPC.execute(namePendingRequest);
+        } catch (const UniValue& e) {
+            UniValue message = find_value( e, "message");
+            LogPrintf ("name_pending lookup error: %s\n", message.get_str().c_str());
```

Comment:
Do you want to `return` here, or is it indeed fine to continue?  If the latter, please add a comment to make this clear (and explain why).

(Reading further, it seems that it is probably fine to continue after an error.  But as written, please add a comment to make this clear to a reader of the code.)